packages/InlineEditing/stories/InlineEditing.stories.mdx

```diff
@@ -0,0 +1,278 @@
+
+<Toast kind={Toast.types.kind.WARNING} hasCloseButton={false}>
+  Work in progress ahead, this component still requires some more work to be ready for production, but the API is
+  stable, so you can start integrating it into your application.
+</Toast>
+
+# Inline Editing
+
+The `<InlineEditing />` component allows you to display two values to the user: one when the component is not in editing mode and an input whenever the user interact and set
+the component into editing mode.
+
+## How does it works?
+
+The InlineEditing package includes a component name `<Editor />` that you can use to encapsulate this kind of interaction. The `<Editor />` Component comes with two
+subcomponents: `<Editor.Value />` and `<Editor.Edit />`. These two components will switch between each other whenever the prop `isEditing` on the `Editor` component changes.
+
+The following is an example of using the Editor component to build a simple Text (value) `->` Text (input) interaction:
+
+<Blocks>
+  <Block>
+    <AppInlineInput />
+  </Block>
+  <Block>
+
+```jsx
+function InlineInput(props) {
+  const { value, onSubmit, isEditing, onClick } = props;
+  const refInput = React.useRef(null);
+
+  function handleClick() {
+    onClick();
+  }
+
+  function submit() {
+    onSubmit({ nextValue: refInput.current.value });
+  }
+
+  return (
+    <Editor onEdit={() => {}} isEditing={isEditing} onClick={handleClick}>
+      <Editor.Edit>
+        {/** we are going to only report back when the user hits Enter */}
+        <Input
+          defaultValue={value}
+          type="text"
+          ref={refInput}
+          onKeyUp={event => {
+            if (event.key === "Enter") {
+              submit();
+            }
+          }}
+        />
+      </Editor.Edit>
+      <Editor.Value>
+        <span style={{ textDecoration: "underline" }}>{value}</span>
+      </Editor.Value>
+    </Editor>
+  );
+}
+
+export function App() {
+  const [value, setValue] = React.useState("Click on it to edit it");
+  const [isEditing, setIsEditing] = React.useState(false);
+
+  function handleClick() {
+    setIsEditing(true);
+  }
+
+  function handleSubmit({ nextValue }) {
+    setValue(nextValue);
+    setIsEditing(false);
+  }
+
+  // prettier-ignore
+  return (
+    <InlineInput 
+      value={value} 
+      onClick={handleClick} 
+      isEditing={isEditing} 
+      onSubmit={handleSubmit} 
+    />
+  )
+}
+```
+
+  </Block>
+</Blocks>
+
+The previous example is a simple attempt to create an InlineEditng Input. To have it fully functional will still require more work with the keyboard interactions,
+add focus onClick, cancel it onKeyUp === "Escape", etc.
+
+And because wiring a fully functionally InlineEditing component can be tedious and tricky to make it work properly, we are currently providing three components ready to use:
+
+- Input type text
+- TextArea
+- ListBox
+
+## Available InlineEditing Components
```

Comment:
> Seems like you definitely do intend to provide these three components to the consumer

Yeap i'm providing those to the consumer but not from an index.js file instead of by `lib/component` option.

Is easier to have all these compose component in the same package, imagine changing one prop or requirement and have to go to `n` packages to update the pacakge.json plus changelog per change. It would just add unnecessary complexity. 

plus

`import ListBox from "@paprika/inline-editing/lib/ListBox"`

make it really clear what are you doing and what are you trying to solve.